src/libraries/Common/src/System/Net/Mail/MailAddressParser.cs

```diff
@@ -104,41 +121,70 @@ private static MailAddress ParseAddress(string data, bool expectMultipleAddresse
                 {
                     index--; // Skip the angle bracket
                     // Skip whitespace, but leave comments, as they may be part of the display name.
-                    index = WhitespaceReader.ReadFwsReverse(data, index);
+                    if (!WhitespaceReader.TryReadFwsReverse(data, index, out index, throwExceptionIfFail))
+                    {
+                        return false;
+                    }
                 }
                 else
                 {
-                    // Mismatched angle brackets, throw
-                    throw new FormatException(SR.Format(SR.MailHeaderFieldInvalidCharacter,
-                        (index >= 0 ? data[index] : MailBnfHelper.EndAngleBracket)));
+                    if (throwExceptionIfFail)
+                    {
+                        // Mismatched angle brackets, throw
```

Comment:
This comment still applies when not throwing, perhaps it should go outside of the if/else.

Reply:
Updated here and also on other place with same issue